com.unity.render-pipelines.high-definition/Runtime/Material/Lit/Lit.hlsl

```diff
@@ -578,15 +578,55 @@ void EncodeIntoGBuffer( SurfaceData surfaceData
+        // 3rd quadrant (Sin <= 0, Cos <  0): AnisoGGX(α, β, γ) == AnisoGGX(α, β, γ + Pi)
+        // 4th quadrant (Sin <  0, Cos >= 0): AnisoGGX(α, β, γ) == AnisoGGX(β, α, γ + Pi * 3/2)
+        // Handling of the interval end-points may be less rigorous to simplify programming.
+        // The only requirement is that the handling is consistent throughout.
+        bool quad2or4 = (sinFrame * cosFrame) < 0;
+
+        // Anisotropy = (α - β) / (α + β).
+        // Exchanging the roughness values α and β is equivalent to negating the value of anisotropy.
+    #if 0
+        // To avoid shading seams at the locations where anisotropy changes its sign,
+        // its magnitude must be the same (on both sides) after reconstruction from the G-buffer.
+        // This means that the hardware unit must perform rounding accurately (and consistently)
+        // before storing the value in the G-buffer.
+        float sfltAniso  = quad2or4 ? -surfaceData.anisotropy : surfaceData.anisotropy;
+        float anisotropy = sfltAniso * 0.5 + 0.5;
+    #else
+        // It turns out, certain hardware has poor rounding behavior:
+        // https://microsoft.github.io/DirectX-Specs/d3d/archive/D3D11_3_FunctionalSpec.htm#3.2.3.6%20FLOAT%20-%3E%20UNORM
+        // Therefore, we must round manually to avoid the seams.
+        float uintAniso  = round(surfaceData.anisotropy * 127.5 + 127.5);
+              uintAniso  = quad2or4 ? 255 - uintAniso : uintAniso;
+        // We cannot represent the anisotropy value of 0 exactly, but it is of little
+        // importance since you can just use the isotropic material for that purpose.
+        float anisotropy = uintAniso * rcp(255);
+    #endif
 
-        outGBuffer2.rgb = float3(surfaceData.anisotropy * 0.5 + 0.5,
-                                 sinOrCos,
-                                 PackFloatInt8bit(surfaceData.metallic, storeSin | quadrant, 8));
+        // We need to convert the values of Sin and Cos to those appropriate for the 1st quadrant.
+        // To go from Q3 to Q1, we must rotate by Pi, so taking the absolute value suffices.
+        // To go from Q2 or Q4 to Q1, we must rotate by ((N + 1/2) * Pi), so we must
+        // take the absolute value and also swap Sin and Cos.
+        bool  storeSin = (abs(sinFrame) < abs(cosFrame)) != quad2or4;
+        // sin [and cos] are approximately linear up to [after] Pi/4 ± Pi.
+        float sinOrCos = min(abs(sinFrame), abs(cosFrame));
+        // To avoid storing redundant angles, we must convert from a node-centered representation
+        // to a cell-centered one, e.i. remap: [0.5/256, 255.5/256] -> [0, 1].
+        float remappedSinOrCos = Remap01(sinOrCos, sqrt(2) * 256.0/255.0, 0.5/255.0);
+
+        outGBuffer2.rgb = float3(anisotropy,
+                                 remappedSinOrCos,
+                                 PackFloatInt8bit(surfaceData.metallic, storeSin ? 1 : 0, 8));
```

Comment:
shouldn't this be ackFloatInt8bit(surfaceData.metallic, storeSin ? 1 : 0, 1)) to save the reamining 2 bit?
(doesn't change the result)

Reply:
we should put a comment

Reply:
I did not change it precisely because it would affect the precision of stored metallic values, to maintain the visuals. Note that other Lit materials also use a 5-bit metallic layout.